lib/teleterm/clusters/storage.go

```diff
@@ -160,7 +159,7 @@ func (s *Storage) addCluster(ctx context.Context, dir, webProxyAddress string) (
 		return nil, trace.Wrap(err)
 	}
 
-	if err := cfg.SaveProfile(false); err != nil {
+	if err := clusterClient.SaveProfile(false); err != nil {
```

Comment:
Back in #20263 I wasn't sure if it's fine to call `SaveProfile` on `TeleportClient` instead of `Config`. Now after adding the test I know that it's fine and it saves us from having to manually set `ClientStore` on `Config` – this is done by `client.NewClient` now, using the shared logic from that function.